plugins/by-name/oil-git-status/default.nix

```
@@ -0,0 +1,68 @@
+{
+  lib,
+  ...
+}:
+lib.nixvim.plugins.mkNeovimPlugin {
+  name = "oil-git-status";
+  packPathName = "oil-git-status.nvim";
+  package = "oil-git-status-nvim";
+
+  description = ''
+    Add Git Status to oil.nvim directory listings.
+    Git status is added to the listing asynchronously after creating the oil directory
+    listing so it won't slow oil down on big repositories. The plugin puts the status in two new sign columns
+    the left being the status of the index, the right being the status of the working directory.
+    This plugin requires lazy loading and the following setting in oil:
+
+    ```(nix)
+    # lazy loading
+    plugins.lz-n.enable = true;
+    plugins.oil-git-status = {
+      lazyLoad = {
+        settings = {
+          enable = true;
+          enabled = true;
+          before.__raw = \'\'
+            function()
+              require("oil").setup({win_options = { singcolumn = "yes:2" }})
+              end
+          \'\';
+        };
+      };
+    };
+  '';
+
+  maintainers = [ lib.maintainers.FKouhai ];
+
```

We may need to set a `configLocation` so that the plugin loads after oil when _not_ lazyloading:

```suggestion

    # Ensure we setup after plugins.oil, which uses the default prio of 1000
    configLocation = lib.mkOrder 1100 "extraConfigLua";

```

---

I'm going to try this out maybe that way I don't need lazy loading, my main thought was okay oil-git-status is loading before oil so I need to lazy load it but I'll try with this thank you very much :)

---

yep this did the trick